File: src/main/java/com/badminton/entity/Member.java
package com.badminton.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("b_member")
public class Member {

    @TableId(type = IdType.AUTO)
    private Integer id;

    private Integer userId;

    private String realName;

    /** 0-未知, 1-男, 2-女 */
    private Integer gender;

    private Integer age;

    private String phone;

    private String avatar;

    private String introduction;

    private Integer level;

    private Integer points;

    private BigDecimal balance;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: Geeker-Admin/src/api/member.ts
import request from '@/utils/request'

export namespace MemberAPI {
  export interface MemberVO {
    id?: number
    userId?: number
    username?: string
    password?: string
    role?: number
    status?: number
    realName?: string
    gender?: number
    age?: number
    phone?: string
    avatar?: string
    introduction?: string
    level?: number
    points?: number
    balance?: number
    createTime?: string
    updateTime?: string
  }

  export interface MemberQuery {
    pageNum: number
    pageSize: number
    realName?: string
    gender?: number
    level?: number
    sortProp?: string
    sortOrder?: string
  }
}

export const MemberApi = {
  getList: (params: MemberAPI.MemberQuery) => {
    const query: Record<string, any> = {
      pageNum: params.pageNum,
      pageSize: params.pageSize,
    }
    if (params.realName) {
      query.realName = params.realName
    }
    if (params.gender !== undefined) {
      query.gender = params.gender
    }
    if (params.level !== undefined) {
      query.level = params.level
    }
    if (params.sortProp) {
      query.sortProp = params.sortProp
    }
    if (params.sortOrder) {
      query.sortOrder = params.sortOrder
    }
    return request.get('/admin/member/page', query)
  },

  getById: (id: number) => {
    return request.get(`/admin/member/${id}`)
  },

  add: (data: MemberAPI.MemberVO) => {
    return request.post('/admin/member', data)
  },

  update: (data: MemberAPI.MemberVO) => {
    return request.put('/admin/member', data)
  },

  delete: (id: number) => {
    return request.delete(`/admin/member/${id}`)
  },
}
